core/state/database.go

```diff
@@ -238,19 +238,57 @@ func (db *cachingDB) purgeLoop() {
 
 // OpenTrie opens the main account trie at a specific root hash.
 func (db *cachingDB) OpenTrie(root common.Hash) (Trie, error) {
+	//try cache first
+	if db.accountTrieCache != nil {
+		if tr, ok := db.accountTrieCache.Get(root); ok {
+			return tr.(*trie.SecureTrie), nil
+		}
+	}
+
 	tr, err := trie.NewStateTrie(trie.StateTrieID(root), db.triedb)
 	if err != nil {
 		return nil, err
 	}
+
+	//update LRU
+	if db.accountTrieCache != nil {
+		db.accountTrieCache.Add(root, tr.ResetCopy())
+	}
```

Comment:
```suggestion

	db.CacheAccount(root, tr)
```

How about re-using `db.CacheAccount()` to update the cache?